test/attr.spec.js

```diff
@@ -177,6 +199,16 @@ describe("Rule: attr", function(){
             "rule::whitelist": true,
         });
     });
+
+    it("should fail in whitelist-mode when all required attributes are allowed but optional are not", function(){
```

Comment:
If I understand correctly this is testing whether optional attributes with mismatched values are handled correctly. If so, I think this title works a bit better:

```suggestion
    it("should fail in whitelist-mode with an invalid value for an optional attribute", function(){
```